resources/js/Pages/Profile.tsx

```diff
@@ -0,0 +1,191 @@
+import {useForm} from '@inertiajs/inertia-react'
+import {
+    Alert,
+    Button,
+    CircularProgress,
+    Divider,
+    Grid,
+    Snackbar,
+    TextField,
+    Typography,
+} from '@mui/material'
+import {ThemeProvider} from '@mui/material/styles'
+import {DeleteModal, ErrorText, Select} from 'Components'
+import {ApplicationLayout} from 'Layouts'
+import React, {useEffect, useState} from 'react'
+import {getFontTheme, useUser} from 'Utils'
+import route from 'ziggy-js'
+
+type UserEditDataType = {
+    name: string
+    email: string
+    password?: string
+    password_confirmation?: string
+    language: string
+}
+
+const Profile = () => {
+    const theme = getFontTheme('Form', 12)
+    const {user} = useUser()
+    const USER_EDIT_INITIAL_VALUE = {
+        name: user.name,
+        email: user.email,
+        password: undefined,
+        password_confirmation: undefined,
+        language: user.language ?? 'en',
+    }
+    const {
+        data,
+        errors,
+        setData,
+        put,
+        reset,
+        wasSuccessful,
+        processing,
+        delete: destroy,
+    } = useForm<UserEditDataType>(USER_EDIT_INITIAL_VALUE)
+    const [isDeleteModalVisible, setIsDeleteModalVisible] = useState<boolean>(false)
+    const [isSuccess, setIsSuccess] = useState<boolean>(wasSuccessful)
+
+    useEffect(() => {
```

Comment:
in case you don't understand what this `useEffect` hook does, basically every time `wasSuccessful` is changed, this hook will set the value of `isSuccess` to `wasSuccessful` and `isSuccess` is the visible state of the `Snackbar` which displays a success message and will automatically change `isSuccess` to `false` after 3 seconds.

Reply:
for future reference, I will refactor this and extract to a method to display notifications most likely.